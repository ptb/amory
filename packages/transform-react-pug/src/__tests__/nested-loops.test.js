import testHelper from './test-helper';

testHelper(__dirname + '/nested-loops.input.js');
