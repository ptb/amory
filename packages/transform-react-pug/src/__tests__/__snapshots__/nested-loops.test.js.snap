// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript output: transformed source code 1`] = `
"module.exports = <div>{((_pug_nodes, _pug_arr) => {
    if (!(_pug_arr == null || Array.isArray(_pug_arr))) throw new Error(\\"Expected \\\\\\"[1,2,3]\\\\\\" to be an array because it is passed to each.\\");
    if (_pug_arr == null || _pug_arr.length === 0) return undefined;

    for (let _pug_index = 0; _pug_index < _pug_arr.length; _pug_index++) {
      let _name;

      const a = _pug_arr[_pug_index];
      _pug_nodes[_pug_nodes.length] = (_name = 'a', undefined);
      _pug_nodes[_pug_nodes.length] = <div key={\\"pug\\" + a + \\":0\\"}>{_name + ':' + a}{((_pug_nodes2, _pug_arr2) => {
          if (!(_pug_arr2 == null || Array.isArray(_pug_arr2))) throw new Error(\\"Expected \\\\\\"[1,2,3]\\\\\\" to be an array because it is passed to each.\\");
          if (_pug_arr2 == null || _pug_arr2.length === 0) return undefined;

          for (let _pug_index2 = 0; _pug_index2 < _pug_arr2.length; _pug_index2++) {
            let _name2;

            const b = _pug_arr2[_pug_index2];
            _pug_nodes2[_pug_nodes2.length] = (_name2 = 'b', undefined);
            _pug_nodes2[_pug_nodes2.length] = <div key={\\"pug\\" + b + \\":0\\"}>{_name2 + ':' + b}</div>;
          }

          return _pug_nodes2;
        })([], [1, 2, 3])}{((_pug_nodes3, _pug_arr3) => {
          if (!(_pug_arr3 == null || Array.isArray(_pug_arr3))) throw new Error(\\"Expected \\\\\\"[1,2,3]\\\\\\" to be an array because it is passed to each.\\");
          if (_pug_arr3 == null || _pug_arr3.length === 0) return undefined;

          for (let i = 0; i < _pug_arr3.length; i++) {
            let _name3;

            const c = _pug_arr3[i];
            _pug_nodes3[_pug_nodes3.length] = (_name3 = 'c', undefined);
            _pug_nodes3[_pug_nodes3.length] = <div key={\\"pug\\" + c + \\":0\\"}>{_name3 + c}</div>;
          }

          return _pug_nodes3;
        })([], [1, 2, 3])}</div>;
    }

    return _pug_nodes;
  })([], [1, 2, 3])}</div>;"
`;

exports[`html output: generated html 1`] = `
<div>
  <div>
    a:1
    <div>
      b:1
    </div>
    <div>
      b:2
    </div>
    <div>
      b:3
    </div>
    <div>
      c1
    </div>
    <div>
      c2
    </div>
    <div>
      c3
    </div>
  </div>
  <div>
    a:2
    <div>
      b:1
    </div>
    <div>
      b:2
    </div>
    <div>
      b:3
    </div>
    <div>
      c1
    </div>
    <div>
      c2
    </div>
    <div>
      c3
    </div>
  </div>
  <div>
    a:3
    <div>
      b:1
    </div>
    <div>
      b:2
    </div>
    <div>
      b:3
    </div>
    <div>
      c1
    </div>
    <div>
      c2
    </div>
    <div>
      c3
    </div>
  </div>
</div>
`;

exports[`static html output: static html 1`] = `"<div><div>a:1<div>b:1</div><div>b:2</div><div>b:3</div><div>c1</div><div>c2</div><div>c3</div></div><div>a:2<div>b:1</div><div>b:2</div><div>b:3</div><div>c1</div><div>c2</div><div>c3</div></div><div>a:3<div>b:1</div><div>b:2</div><div>b:3</div><div>c1</div><div>c2</div><div>c3</div></div></div>"`;
